File: Maths_And_Stats/Number_Theory/Ugly_Number/Ugly_Number.cpp
#include <iostream>

bool IsUglyNumber(int num)
{
	auto MaxDivide = [=](int& left, const int& right) {
		while (left % right == 0) left = left / right;
		return left;
	};

	num = MaxDivide(num, 2);
	num = MaxDivide(num, 3);
	num = MaxDivide(num, 5);

	return num == 1;
}

int main()
{
	for (int i = 1; i < 50; ++i)
	{
		printf("%d is ugly? %d\n", i, IsUglyNumber(i));
	}

	return EXIT_SUCCESS;
}

File: Maths_And_Stats/Number_Theory/Ugly_Number/Ugly_Number.py
#Function checks if input is an ugly number or not
import sys

#Function from: w3resource.com
def is_ugly_number(num):
    if num == 0:
        return False
    for i in [2, 3, 5]: #2,3 and 5 must be the only prime factors of the ugly number
        while num % i == 0:
            num /= i
    return num == 1

print(is_ugly_number(int(sys.argv[1])))
